hubstorage/client.py

```diff
@@ -21,22 +42,48 @@ class HubstorageClient(object):
     DEFAULT_TIMEOUT = 60.0
 
     def __init__(self, auth=None, endpoint=None, connection_timeout=None,
-            max_retries=0):
+            max_retries=3):
         self.auth = xauth(auth)
         self.endpoint = endpoint or self.DEFAULT_ENDPOINT
         self.connection_timeout = connection_timeout or self.DEFAULT_TIMEOUT
-        self.session = self._create_session(max_retries)
+        self.session = self._create_session()
+        self.retrier = Retrying(stop_max_attempt_number=max_retries + 1, retry_on_exception=_hc_retry_on_exception)
```

Comment:
I think that makes things too complex, both for us and the user.

What is a useful value for `max_delay`?
- `max_retries` * 50s (haproxy timeout) + exponential backoff? so more than 150s in the default case?
- the current default? 60s, so that queries that timeout (the reason for this patch) are retried only once?
- if the user sets only `max_retry` to a new value, should we change the `max_delay` to accommodate it?
- if the user sets both max_retry and max_delay, should we warn when they are incompatible?
- if the user sets max_delay to a small value like 10s should we warn?

AFAIK the library is used for long running jobs so no one wants to fail sooner with an error rather than trying to finish the job.

Setting an exponential multiplier and a jitter for the sake of hubstorage/jobq is a good idea, but there's no reason to expose these 2 options to the user, we own the API.

Something like 500ms for the exponential backoff? (wait 1, 2 then 4 seconds between requests).
And a jitter to something like 500ms?